util/ipgen/renderer.py

```diff
@@ -108,13 +125,12 @@ def _tplfunc_instance_vlnv(self, template_vlnv_str: str) -> str:
         template_core_version = (template_vlnv[3]
                                  if len(template_vlnv) == 4 else None)
 
-        # Remove the template name from the start of the core name.
-        # For example, a core name `rv_plic_component` will result in an
-        # instance name 'my_instance_component' (instead of
-        # 'my_instance_rv_plic_component').
         if "module_instance_name" in self.ip_config.param_values:
-            assert template_core_name.startswith(
-                self.ip_config.param_values["module_instance_name"])
+            if not template_core_name.startswith(
+                    self.ip_config.param_values["module_instance_name"]):
+                raise TemplateRenderError(
+                    f'The name field in {template_vlnv_str} must start with '
+                    'self.ip_config.param_values["module_instance_name"]')
```

Comment:
Don't you want to insert the value of this module instance name here, rather than the name of the Python variable?

Reply:
Certainly, good catch.